### PR TITLE
Jetpack Social: Update republicize plan check

### PR DESCRIPTION
**Pull request by spsiddarthan**
Remove plan checks for republicize and make it free. Currently you need jetpack security or higher for republicize and on the wpcom side, you need a premium plan. This change makes republicize free. 
#### Changes proposed in this Pull Request:

* Removed plan checks for republicize
* Cleaned up `use-publicize-config`

#### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Jetpack product discussion

p1661262228191369/1661236418.340299-slack-C02JJ910CNL
#### Does this pull request change what data or activity we track or use?

NO
#### Testing instructions:

* Be able to publicize on jetpack sites that have a free plan, backup plan and a security plan
* Be able to republicize on wpcom sites that are on a free plan (simple plan) and premium plan (after sandboxing the phab patch)

### CODE REVIEW

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 1]
```diff
@@ -13,14 +13,13 @@ import {
...
 } from '@automattic/jetpack-publicize-components';
-import { PanelBody, PanelRow, ToggleControl, Disabled } from '@wordpress/components';
+import { PanelBody, PanelRow, ToggleControl } from '@wordpress/components';
 import { useSelect } from '@wordpress/data';
 import { store as editorStore } from '@wordpress/editor';
 import { Fragment } from '@wordpress/element';
 import { __ } from '@wordpress/i18n';
 import { SharePostRow } from '../../components/share-post';
 import usePublicizeConfig from '../../hooks/use-publicize-config';
-import UpsellNotice from '../upsell';
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js:23]: If we're not using the upsell, can we delete the entire component as well?

File: projects/plugins/jetpack/changelog/y [hunk 2]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: enhancement
+
+Remove plan checks for republicize.
```
Comment (danielpost) [inline projects/plugins/jetpack/changelog/y]: I think the file name is wrong here.

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 3]
```diff
@@ -30,11 +29,9 @@ const PublicizePanel = ( { prePublish } ) => {
 		isRePublicizeFeatureEnabled,
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Does it make sense to still have this boolean? It's used in the [Publicize Form](https://github.com/Automattic/jetpack/blob/trunk/projects/js-packages/publicize-components/src/components/form/index.js) but we can probably get rid of the usage there as well.

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 4]
```diff
@@ -48,30 +45,25 @@ const PublicizePanel = ( { prePublish } ) => {
...
-	const PanelRowWithDisabled = isPublicizeDisabledBySitePlan ? Disabled : PanelRow;
-
 	// Panel wrapper.
 	const PanelWrapper = prePublish ? Fragment : PanelBody;
 	const wrapperProps = prePublish
 		? {}
 		: {
 				title: __( 'Share this post', 'jetpack' ),
-				className: isPublicizeDisabledBySitePlan ? 'jetpack-publicize-disabled' : '',
+				className: '',
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Can we just get rid of the `className` prop completely?

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 5]
```diff
@@ -48,30 +45,25 @@ const PublicizePanel = ( { prePublish } ) => {
...
 		  };
 
 	return (
 		<PanelWrapper { ...wrapperProps }>
-			<UpsellNotice isPostPublished={ isPostPublished } />
-
-			{ ! hideRePublicizeFeature && (
+			{
 				<Fragment>
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: I think the `{}` and `<Fragment>` are now redundant.

File: projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js [hunk 6]
```diff
@@ -33,31 +30,14 @@ export default function usePublicizeConfig() {
...
-		isPublicizeEnabledMeta;
-
-	/*
-	 * isPublicizeDisabledBySitePlan:
-	 * Depending on the site plan and type, the republicize feature
-	 * would get dissabled.
-	 */
-	const isPublicizeDisabledBySitePlan =
-		isRePublicizeFeatureEnabled && isPostPublished && isRePublicizeUpgradableViaUpsell;
+		( isPostPublished && ! isRePublicizeFeatureEnabled ) || isPublicizeEnabledMeta;
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js]: Does this still makes sense? In the former check, it was enabled if the post was published AND the user didn't have Republicize—now it seems like Publicize gets disabled for published posts, considering `isRePublicizeFeatureEnabled` should always be true? Can we get rid of this check altogether?

File: projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js [hunk 7]
```diff
@@ -68,19 +48,14 @@ export default function usePublicizeConfig() {
 	 * it needs to hide part of the Publicize feature.
 	 */
 	const hideRePublicizeFeature =
-		isPostPublished &&
-		! isRePublicizeFeatureAvailable &&
-		! isRePublicizeUpgradableViaUpsell &&
-		isRePublicizeFeatureEnabled;
+		isPostPublished && ! isRePublicizeFeatureAvailable && isRePublicizeFeatureEnabled;
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js]: Should we get rid of this as well?

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 4 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Done!

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 5 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Sure!

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 3 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: I think the publicize module can be disabled, which is why I think this exists. Let me experiment with that and reply to this comment.

File: projects/plugins/jetpack/changelog/y (hunk 2 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/changelog/y]: Will fix

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 1 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js:23]: Fusion wasn't building the phab patch when I did that, so I can remove the mapping and the file in a follow up PR.

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 3 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: This had nothing to do with that, I have removed this boolean

File: projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js (hunk 7 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js]: Done!

File: projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js (hunk 6 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/hooks/use-publicize-config/index.js]: done!

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 8]
```diff
@@ -48,55 +42,47 @@ const PublicizePanel = ( { prePublish } ) => {
...
-										  )
-								}
-								onChange={ togglePublicizeFeature }
-								checked={ isPublicizeEnabled }
-								disabled={ ! hasConnections }
-							/>
-						</PanelRowWithDisabled>
-					) }
+			<Fragment>
+				{ ! isPostPublished && (
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Do we need the `<Fragment>` here?

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js [hunk 9]
```diff
@@ -48,55 +42,47 @@ const PublicizePanel = ( { prePublish } ) => {
...
-						isPublicizeEnabled={ isPublicizeEnabled }
-						isRePublicizeFeatureEnabled={ isRePublicizeFeatureEnabled }
-						isPublicizeDisabledBySitePlan={ isPublicizeDisabledBySitePlan }
-					/>
-					{ isPublicizeEnabled && <PublicizeTwitterOptions prePublish={ prePublish } /> }
+				<PublicizeConnectionVerify />
+				<PublicizeForm
+					isPublicizeEnabled={ isPublicizeEnabled }
+					isRePublicizeFeatureEnabled={ true }
+					isPublicizeDisabledBySitePlan={ false }
```
Comment (danielpost) [inline projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: We can probably get rid of these altogether and clean up the `PublicizeForm` component, but we can do that in a separate PR.

**Review by danielpost — DISMISSED** [review]
Nice work and tests well! I added some questions but looks good to me 👍🏼

File: projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js (hunk 9 above)
Comment (spsiddarthan) [reply projects/plugins/jetpack/extensions/plugins/publicize/components/panel/index.js]: Sure thing!

**Review by danielpost — APPROVED** [review]

**Review by coder-karen — APPROVED** [review]
Works well for me. 👍 

I did get an 'Unable to share post' error message after attempting to Republicize to two accounts, but the post was still shared (to one of the two accounts). That's not relevant for this PR though - I think more for #25661.

**Comment by spsiddarthan** [comment]
Thanks @coder-karen! We had a team call today where we realized that we are going to land this after we build republicize in the social plugin first, so this PR may not get merged for a bit.

**Comment by spsiddarthan** [comment]
I am closing this PR to avoid it being merged accidentally.
